tests/primeqa/services/rest_server/test_server.py

```diff
@@ -0,0 +1,183 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright 2022 PrimeQA Team
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import List
+from unittest.mock import MagicMock
+import pytest
+from fastapi import FastAPI
+from fastapi.testclient import TestClient
+# from primeqa.services.rest_server.server import FastAPI
+import sys
+import os
+from primeqa.services.configurations import Settings
+from primeqa.services.rest_server.data_models import Reader
+from primeqa.services.rest_server.server import app
+
+
+class TestServer:
+
+    @pytest.fixture()
+    def client(self):
+        return TestClient(app)
+
+    @pytest.fixture()
+    def mock_STORE(self, mocker) -> MagicMock:
+        return mocker.patch(
+            "primeqa.services.rest_server.server.STORE",
+            autospec=True,
+        )
+
```

Comment:
I would recommend mocking registries like `ReadersRegistry`, `RetriversRegistry` etc. and testing that appropriate function on those registries are called. That way we aren't testing Registry functionalities in this unit test.